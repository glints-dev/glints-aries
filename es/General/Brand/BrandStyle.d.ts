export declare const BrandContainer: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const BrandImage: import("styled-components").StyledComponent<"img", any, {}, never>;
