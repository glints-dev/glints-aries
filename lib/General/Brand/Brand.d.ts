import * as React from 'react';
import { BrandContainer } from './BrandStyle';
declare const Brand: ({ alt, asset, className, rightClickURL, onContextMenu, ...defaultProps }: Props) => JSX.Element;
export declare type Props = React.ComponentPropsWithoutRef<typeof BrandContainer> & {
    /** There are 3 available values for asset props. The first and second one are Glints logo, but you can also custom the logo by parsing url or path of the logo. */
    asset: string;
    /** HTML <img> alt Attribute */
    alt?: HTMLImageElement['alt'];
    className?: string;
    /** Redirect to url when right click is triggered. */
    rightClickURL?: string;
    /** Executes when the user right-clicks on the component */
    onContextMenu?(): void;
};
export default Brand;
