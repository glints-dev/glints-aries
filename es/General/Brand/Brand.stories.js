import React from 'react';
import { Brand } from './index';
import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
export default {
  title: 'General/Brand',
  component: Brand,
  argTypes: {
    asset: {
      control: {
        type: 'radio',
        options: ['glints-black', 'glints-white']
      }
    },
    className: {
      control: {
        disable: true
      }
    },
    onContextMenu: {
      control: {
        disable: true
      }
    }
  },
  decorators: [function (Story) {
    return /*#__PURE__*/React.createElement(BaseContainer, null, Story());
  }]
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Brand, args);
};

export var Interactive = Template.bind({});
Interactive.args = {
  asset: 'glints-black'
};
export var RedirectToGlintsWhenRightClick = Template.bind({});
RedirectToGlintsWhenRightClick.args = {
  asset: 'glints-black',
  rightClickURL: 'https://glints.com'
};