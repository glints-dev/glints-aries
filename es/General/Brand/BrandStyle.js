import styled from 'styled-components';
export var BrandContainer = styled.div.withConfig({
  displayName: "BrandStyle__BrandContainer",
  componentId: "sc-xckggx-0"
})(["position:relative;display:inline-flex;cursor:pointer;&:focus{outline:none;}&:focus > img{outline:5px auto -webkit-focus-ring-color;}"]);
export var BrandImage = styled.img.withConfig({
  displayName: "BrandStyle__BrandImage",
  componentId: "sc-xckggx-1"
})(["object-fit:contain;width:3em;height:3em;outline:none;"]);