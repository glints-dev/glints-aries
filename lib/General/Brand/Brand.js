"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var React = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _BrandStyle = require("./BrandStyle");

var _excluded = ["alt", "asset", "className", "rightClickURL", "onContextMenu"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var GlintsBlack = require('../../../assets/image/glints-logo-black.svg');

var GlintsWhite = require('../../../assets/image/glints-logo-white.svg');

var Brand = function Brand(_ref) {
  var alt = _ref.alt,
      asset = _ref.asset,
      className = _ref.className,
      rightClickURL = _ref.rightClickURL,
      onContextMenu = _ref.onContextMenu,
      defaultProps = (0, _objectWithoutPropertiesLoose2["default"])(_ref, _excluded);

  var handleRightClick = function handleRightClick(e) {
    if (rightClickURL) {
      e.preventDefault();

      if (onContextMenu !== undefined) {
        onContextMenu();
      }

      window.location.href = rightClickURL;
    }
  };

  var srcAsset = '';

  if (asset === 'glints-black') {
    srcAsset = GlintsBlack;
  } else if (asset === 'glints-white') {
    srcAsset = GlintsWhite;
  } else {
    srcAsset = asset;
  }

  return /*#__PURE__*/React.createElement(_BrandStyle.BrandContainer, (0, _extends2["default"])({
    className: (0, _classnames["default"])('aries-brand', className),
    role: "presentation",
    tabIndex: 0,
    onContextMenu: handleRightClick
  }, defaultProps), /*#__PURE__*/React.createElement(_BrandStyle.BrandImage, {
    className: "brand-image",
    src: srcAsset,
    alt: alt,
    tabIndex: -1
  }));
};

var _default = Brand;
exports["default"] = _default;