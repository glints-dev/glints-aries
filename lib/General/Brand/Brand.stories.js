"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports["default"] = exports.RedirectToGlintsWhenRightClick = exports.Interactive = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("./index");

var _GlintsContainer = require("../../Layout/GlintsContainer/GlintsContainer");

var _default = {
  title: 'General/Brand',
  component: _index.Brand,
  argTypes: {
    asset: {
      control: {
        type: 'radio',
        options: ['glints-black', 'glints-white']
      }
    },
    className: {
      control: {
        disable: true
      }
    },
    onContextMenu: {
      control: {
        disable: true
      }
    }
  },
  decorators: [function (Story) {
    return /*#__PURE__*/_react["default"].createElement(_GlintsContainer.BaseContainer, null, Story());
  }]
};
exports["default"] = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_index.Brand, args);
};

var Interactive = Template.bind({});
exports.Interactive = Interactive;
Interactive.args = {
  asset: 'glints-black'
};
var RedirectToGlintsWhenRightClick = Template.bind({});
exports.RedirectToGlintsWhenRightClick = RedirectToGlintsWhenRightClick;
RedirectToGlintsWhenRightClick.args = {
  asset: 'glints-black',
  rightClickURL: 'https://glints.com'
};