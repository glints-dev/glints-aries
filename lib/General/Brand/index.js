"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports["default"] = void 0;

var _Brand = _interopRequireDefault(require("./Brand"));

exports.Brand = _Brand["default"];
var _default = _Brand["default"];
exports["default"] = _default;