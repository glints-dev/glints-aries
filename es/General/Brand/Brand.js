import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/objectWithoutPropertiesLoose";
var _excluded = ["alt", "asset", "className", "rightClickURL", "onContextMenu"];
import * as React from 'react';
import classNames from 'classnames';
import { BrandContainer, BrandImage } from './BrandStyle';

var GlintsBlack = require('../../../assets/image/glints-logo-black.svg');

var GlintsWhite = require('../../../assets/image/glints-logo-white.svg');

var Brand = function Brand(_ref) {
  var alt = _ref.alt,
      asset = _ref.asset,
      className = _ref.className,
      rightClickURL = _ref.rightClickURL,
      onContextMenu = _ref.onContextMenu,
      defaultProps = _objectWithoutPropertiesLoose(_ref, _excluded);

  var handleRightClick = function handleRightClick(e) {
    if (rightClickURL) {
      e.preventDefault();

      if (onContextMenu !== undefined) {
        onContextMenu();
      }

      window.location.href = rightClickURL;
    }
  };

  var srcAsset = '';

  if (asset === 'glints-black') {
    srcAsset = GlintsBlack;
  } else if (asset === 'glints-white') {
    srcAsset = GlintsWhite;
  } else {
    srcAsset = asset;
  }

  return /*#__PURE__*/React.createElement(BrandContainer, _extends({
    className: classNames('aries-brand', className),
    role: "presentation",
    tabIndex: 0,
    onContextMenu: handleRightClick
  }, defaultProps), /*#__PURE__*/React.createElement(BrandImage, {
    className: "brand-image",
    src: srcAsset,
    alt: alt,
    tabIndex: -1
  }));
};

export default Brand;