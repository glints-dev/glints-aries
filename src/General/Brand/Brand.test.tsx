import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import { render, fireEvent } from '@testing-library/react';

import Brand from './Brand';

describe('<Brand/>', () => {
  it('should render as expected', () => {
    const blackLogoSnapshot = renderer
      .create(<Brand asset="glints-black" />)
      .toJSON();
    expect(blackLogoSnapshot).toMatchSnapshot();
    const white = renderer.create(<Brand asset="glints-white" />).toJSON();
    expect(white).toMatchSnapshot();
    const cutsom = renderer
      .create(<Brand asset="http://example.com/example.jpg" />)
      .toJSON();
    expect(cutsom).toMatchSnapshot();
  });

  it('should open the given rightClickUrl', () => {
    delete window.location;
    window.location = {} as Location;

    const url = 'https://glints.com';
    const { getByRole } = render(
      <Brand asset="glints-black" rightClickURL={url} />
    );
    const container = getByRole('presentation');
    fireEvent.contextMenu(container);
    expect(window.location.href).toBe(url);
  });

  it('should call onContextMenu', () => {
    delete window.location;
    window.location = {} as Location;

    const url = 'https://glints.com';
    const onContextMenu = jest.fn();
    const { getByRole } = render(
      <Brand
        asset="glints-black"
        rightClickURL={url}
        onContextMenu={onContextMenu}
      />
    );
    const container = getByRole('presentation');
    expect(onContextMenu).toHaveBeenCalledTimes(0);
    fireEvent.contextMenu(container);
    expect(onContextMenu).toHaveBeenCalledTimes(1);
  });
});
