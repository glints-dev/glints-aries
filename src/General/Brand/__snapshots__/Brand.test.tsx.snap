// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brand/> should render as expected 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c0:focus > img {
  outline: 5px auto -webkit-focus-ring-color;
}

.c1 {
  object-fit: contain;
  width: 3em;
  height: 3em;
  outline: none;
}

<div
  className="c0 aries-brand"
  onContextMenu={[Function]}
  role="presentation"
  tabIndex={0}
>
  <img
    className="c1 brand-image"
    src="test-file-stub"
    tabIndex={-1}
  />
</div>
`;

exports[`<Brand/> should render as expected 2`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c0:focus > img {
  outline: 5px auto -webkit-focus-ring-color;
}

.c1 {
  object-fit: contain;
  width: 3em;
  height: 3em;
  outline: none;
}

<div
  className="c0 aries-brand"
  onContextMenu={[Function]}
  role="presentation"
  tabIndex={0}
>
  <img
    className="c1 brand-image"
    src="test-file-stub"
    tabIndex={-1}
  />
</div>
`;

exports[`<Brand/> should render as expected 3`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c0:focus > img {
  outline: 5px auto -webkit-focus-ring-color;
}

.c1 {
  object-fit: contain;
  width: 3em;
  height: 3em;
  outline: none;
}

<div
  className="c0 aries-brand"
  onContextMenu={[Function]}
  role="presentation"
  tabIndex={0}
>
  <img
    className="c1 brand-image"
    src="http://example.com/example.jpg"
    tabIndex={-1}
  />
</div>
`;
